pkg/cache/observability/logger.go

```diff
@@ -0,0 +1,236 @@
+}
+
+func NewStdLogger() Logger {
+	return &styledLogger{
+		useColors: isTerminal(),
+	}
+}
+
+func (l *styledLogger) getTraceString(ctx context.Context) string {
+	if ctx == nil {
+		return ""
+	}
+
+	sc := trace.SpanFromContext(ctx).SpanContext()
+	if sc.IsValid() {
+		return " " + l.applyColor(Gray, sc.TraceID().String())
+	}
+
+	return ""
+}
+
+func (l *styledLogger) Errorf(ctx context.Context, format string, args ...any) {
+	l.logSimple(ctx, ERROR, Red, format, args...)
+}
+
+func (l *styledLogger) Warnf(ctx context.Context, format string, args ...any) {
+	l.logSimple(ctx, WARN, Yellow, format, args...)
+}
+
+func (l *styledLogger) Infof(ctx context.Context, format string, args ...any) {
+	l.logSimple(ctx, INFO, Green, format, args...)
+}
+
+func (l *styledLogger) Debugf(ctx context.Context, format string, args ...any) {
+	l.logSimple(ctx, DEBUG, Gray, format, args...)
+}
+
+func (l *styledLogger) Hitf(ctx context.Context, _ string, duration time.Duration, operation string) {
+	l.LogRequest(ctx, INFO, "Cache hit", "HIT", duration, operation)
+}
+
+func (l *styledLogger) Missf(ctx context.Context, _ string, duration time.Duration, operation string) {
+	// A miss isn't an error, but we'll color its tag yellow for attention.
+	l.LogRequest(ctx, INFO, "Cache miss", "MISS", duration, operation)
+}
+
+func (l *styledLogger) LogRequest(ctx context.Context, level, message string, tag any, duration time.Duration, operation string) {
+	const tagColumnStart = 45
+
+	const durationColumnStart = 60
+
+	levelStr, levelColor := getLevelStyle(level)
+	ts := l.applyColor(Gray, "["+time.Now().Format(time.TimeOnly)+"]")
+	traceStr := l.getTraceString(ctx)
+	initialPart := fmt.Sprintf("%s %s%s %s", l.applyColor(levelColor, levelStr), ts, traceStr, message)
+
+	tagStr := l.formatTag(tag)
+	durationStr := l.applyColor(Gray, fmt.Sprintf("%dµs", duration.Microseconds()))
+
+	padding1 := getPadding(tagColumnStart, len(stripAnsi(initialPart)))
+	padding2 := getPadding(durationColumnStart, tagColumnStart+len(stripAnsi(tagStr)))
+
+	fmt.Printf("%s%s%s%s%s %s\n",
+		initialPart,
+		padding1,
+		tagStr,
+		padding2,
+		durationStr,
+		operation,
+	)
+}
+
+func (l *styledLogger) logSimple(ctx context.Context, level, color, format string, args ...any) {
+	levelStr := l.applyColor(color, level)
+	ts := l.applyColor(Gray, "["+time.Now().Format(time.TimeOnly)+"]")
+	traceStr := l.getTraceString(ctx)
+	msg := fmt.Sprintf(format, args...)
+	fmt.Printf("%s %s%s %s\n", levelStr, ts, traceStr, msg)
+}
+
+func getLevelStyle(level string) (levelStr, color string) {
+	switch level {
+	case ERROR:
+		return "ERROR", Red
+	case WARN:
+		return "WARN", Yellow
+	case INFO:
+		return "INFO", Green
+	case DEBUG:
+		return "DEBUG", Gray
+	default:
+		return level, Reset
+	}
+}
+
+func (l *styledLogger) formatTag(tag any) string {
+	switch t := tag.(type) {
+	case int:
+		return l.formatIntTag(t)
+	case string:
+		return l.formatStringTag(t)
+	default:
+		return fmt.Sprintf("%v", tag)
```

Comment:
No formatting is used, you can use fmt.Sprint directly

```suggestion
		return fmt.Sprint(tag)
```